README.md

```diff
@@ -467,21 +467,78 @@ That also covers gdb-based GUI - e.g. Visual Studio Code `Debug`.
+
+You can also have km's gdb server listening for connections in the background by using the km \-\-gdb-listen command line flag.
+With this flag the payload begins running immediately and the gdb server listens for gdb client connection while the payload runs.
+You can connect to the gdb server, disconnect, and reconnect as often as you want until the payload completes.
+When you connect to the km gdb server all payload threads will be paused until the gdb client starts them using the "cont" or "step" or "next" commands.
+
+You can only connect to a payload's gdb server from the machine hosting the payload.
+
+km gdb server testing has been done using gdb client with version "GNU gdb (GDB) Fedora 9.1-5.fc32".
+
+The km implementation uses one payload signal to pause payload threads while the gdb client is stopping the payload.  That is signal 63.
+To avoid seeing gdb client messages about this signal it is a good idea to either manually run the "handle SIG63 nostop" command for each debugging
+session or add this command to your ~/.gdbinit file.
+
+If your payload uses fork then the child payload resets all of the gdb related settings so the child payload can't be debugged.  We are working on correcting this shortfall.
+When the child payload finally exec's to a new payload, no gdb related arguments are added to the invocation of km.
+So, there is no debugging in the exec'ed payload either.  We are working on this too.
+
+#### Known gdb related issues
+
+Stack traces are sometimes incomplete.
+gdb non-stop mode (set non-stop on) is not supported.
+Do not alter the GS segment selector in your programs.  It is used internally by km.
+The "info registers" command does not display the contents of the floating point registers.
+The segment register contents are not displayed correctly.
+When debugging payloads with dynamic linking, the km gdb server consumes one payload fd for each library.  Disconnecting the gdb client causes the fd's to be closed and freed.
+
+#### km gdb example
+
+When starting a payload with gdb debugging enabled you would do the following and expect to see the following lines dispalyed by km.
 
 ```
-/opt/kontain/bin/km -g ./cpython/python.km
-./cpython/python.km: Waiting for a debugger. Connect to it like this:
-        gdb --ex="target remote localhost:2159" ./cpython/python.km
+[someone@work ~]$ /opt/kontain/bin/km -g ./tests/hello_test.km
+./tests/hello_test.km: Waiting for a debugger. Connect to it like this:
+        gdb --ex="target remote localhost:2159" ./tests/hello_test.km
 GdbServerStubStarted
 ```
 
-If you want to run payload and connect gdb to it in-flight, pass `-gdb-listen` to KM.
-
-**TODO**: more documentation and examples here.
+You would then run the following to attach the gdb client to the payload debugger:
+
+```
+[someone@work ~]$ gdb --ex="target remote localhost:2159" ./tests/hello_test.km
+GNU gdb (GDB) Fedora 9.1-5.fc32
```

Comment:
Please add `-q` flag to gdb , all the licnes/help output from gdb here does not add any value

Reply:
This is showing what km tells them to do.  km does not tell them to use -q.
Maybe it should be added to what km instructs the user to do.

Reply:
yes, pls add it in the both places , but for now let's drop it here - it just distracting to have so much to print out